assets/js/base/components/payment-methods/express-checkout.js

```diff
@@ -50,6 +68,14 @@ const ExpressCheckoutFormControl = () => {
 		return null;
 	}
 
+	if ( isCompact ) {
+		return (
+			<CompactExpressCheckoutContainer>
+				<ExpressPaymentMethods />
+			</CompactExpressCheckoutContainer>
+		);
+	}
```

Comment:
The breakdown into different components here feels like a patch, rather than logical components. I'm guessing this is complex because we want to share the notices container?

From the high level we have two components we need: 

- `CartExpressPayment` .. or `CartSubmitButtons` (encompassing both express pay and proceed to cart)
- `CheckoutExpressPayment`

The `isCompact` prop and the two variations of `[Compact]ExpressCheckoutContainer` don't feel like a good match for the UI.

I don't think this is a blocker, but if you think we can make these components easier to understand by refactoring, go for it :)

Related – I'm guessing some of this is intended to be used elsewhere, e.g. express payment on product page. Great if we can handle that variation now, but I'd be wary of generalising too early.

Reply:
In a818772f9fb970a705e3dc6427c9ca53b93eac6d I split `ExpressCheckoutFormControl` into two components, one to be used in the Cart block and the other in the Checkout block. Let me know how it looks. :slightly_smiling_face:

Reply:
Looks good to me – naming them based on cart/checkout is much more direct. Can abstract out more later as needed.

I see the class names still use the `compact` naming rather than cart/checkout classes. Is this intentional? I'd assume you'd either style it based on a container class (e.g. `.*cart*`), or the class name would align with component name (e.g. `wc-block-components-express-payment wc-block-components-express-payment__cart` - use BEM modifier for cart vs checkout?)

Up to you how you handle this - happy with whatever you decide.

Reply:
Good point. I was trying to avoid breaking old class names, but probably this is a good opportunity to update them and reorganize them a bit. I did it in 88be9aa2757db307ada24d29c0bbb09f2dc16bc4.